LambdaLinq/Controllers/HomeController.cs

```diff
@@ -31,16 +33,35 @@ public ActionResult Search()
         }
 
         [HttpPost]
-        public ActionResult Search(Book searchBook)
+        public ActionResult Search(SearchViewModel searchBook)
         {
+            
             //TODO: add logic to return all books if the search is empty
-
-            var foundBooks = db.Books.Where(b => b.Author == searchBook.Author);
-
+            if (searchBook.Title == null
+                && searchBook.Author == null)
+            {
+                return View("SearchResult", db.Books);
+            }
+            
+           var foundBooks = db.Books.Where(b => b.Author == searchBook.Author);
+            if (searchBook.Author != null)
+            {
+                return View("SearchResult", foundBooks);
+            }
             //TODO: add logic to search by Title (Note: you will need to adjust the View and ViewModel)
+            var searchTitle = db.Books.Where(b => b.Title == searchBook.Title);
+               if(searchBook.Title != null)
+            {
+                return View("SearchResult", searchTitle);
+            }
 
             //TODO: add logic to return a search on price between a low and high number (Note: you will need to adjust the View and ViewModel)
 
+            var searchPrice = db.Books.Where(b => b.Price >= searchBook.LowP && b.Price <= searchBook.HighP);
+            if (searchBook.LowP > 0 || searchBook.HighP > 0)
```

Comment:
better test would be
if (searchBook.LowP>0 && searchBook.HighP>searchBook.LowP)